Build error because of -Werror=template-id-cdtor in C++20

### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Build completes without errors.

### Current behavior

Build breaks with `thirdparty/filewatch/FileWatch.hpp:129:22: error: template-id not allowed for constructor in C++20 [-Werror=template-id-cdtor]`.

### Steps to reproduce

Build with gcc (GCC) 14.2.1 20240910.

### Fast DDS version/commit

3.1.0 and later for sure.

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

_No response_

### XML configuration file

_No response_

### Relevant log output

_No response_

### Network traffic capture

_No response_

--- a/thirdparty/filewatch/FileWatch.hpp
+++ b/thirdparty/filewatch/FileWatch.hpp
@@ -126,7 +126,7 @@ namespace filewatch {
         FileWatch<T>& operator=(const FileWatch<T>& other) = delete;
 
         // Const memeber varibles don't let me implent moves nicely, if moves are really wanted std::unique_ptr should be used and move that.
-        FileWatch<T>(FileWatch<T>&&) = delete;
+        FileWatch(FileWatch<T>&&) = delete;
         FileWatch<T>& operator=(FileWatch<T>&&) & = delete;
 
     private:

